ParallelMatrixMultiplication/ParallelMatrixMultiplication.Tests/matrixMultiplicationTest.cs

```diff
@@ -0,0 +1,195 @@
+// <copyright file="matrixMultiplicationTest.cs" company="Sarapulov Vasilii">
+// Copyright (c) Sarapulov Vasilii. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+
+namespace ParallelMatrixMultiplication.Tests;
+
+using MatrixGenerator;
+
+/// <summary>
+/// Test class.
+/// </summary>
+public class Tests
+{
+    /// <summary>
+    /// Indexer testing.
+    /// </summary>
+    [Test]
+    public void TestIndexer()
+    {
+        var matrix = new Matrix("../../../TestFiles/matrix1.txt");
+        int k = 1;
+        for (int i = 0; i < 3; i++)
+        {
+            for (int j = 0; j < 3; j++)
+            {
+                Assert.That(matrix[i, j], Is.EqualTo(k));
+                k++;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Matrix equivalence test.
+    /// </summary>
+    /// <param name="inputFilePath">Path to the matrix file.</param>
+    [TestCase("../../../TestFiles/matrix1.txt")]
+    [TestCase("../../../TestFiles/matrix2.txt")]
+    [TestCase("../../../TestFiles/matrix3.txt")]
+    public void TestEquals(string inputFilePath)
+    {
+        Assert.IsTrue(Matrix.Equals(new Matrix(inputFilePath), new Matrix(inputFilePath)));
+    }
+
+    /// <summary>
+    /// Matrix equivalence test.
+    /// </summary>
+    /// <param name="inputFilePath1">Path to the first matrix file.</param>
+    /// <param name="inputFilePath2">Path to the second matrix file.</param>
+    [TestCase("../../../TestFiles/matrix1.txt", "../../../TestFiles/matrix2.txt")]
+    [TestCase("../../../TestFiles/matrix2.txt", "../../../TestFiles/matrix3.txt")]
+    [TestCase("../../../TestFiles/matrix3.txt", "../../../TestFiles/matrix1.txt")]
+    [TestCase("../../../TestFiles/matrix1.txt", "../../../TestFiles/matrix4.txt")]
+    public void TestNotEquals(string inputFilePath1, string inputFilePath2)
+    {
+        Assert.IsFalse(Matrix.Equals(new Matrix(inputFilePath1), new Matrix(inputFilePath2)));
+    }

```

Comment:
```suggestion
    public void TestNotEquals(string inputFilePath1, string inputFilePath2)
        => Assert.IsFalse(Matrix.Equals(new Matrix(inputFilePath1), new Matrix(inputFilePath2)));
```
Вообще, позабавила старательная проверка аж на четырёх наборах входных данных, что матрицы НЕ равны.